### PR TITLE
Add EC_GROUP_to_params to convert an EC_GROUP to an array of OSSL_PARAM

### PR DESCRIPTION
**Pull request by Basskrapfen**
As already shortly discussed in openssl-users there is no reverse operation of
EC_GROUP_new_from_params().
In the appended patch a new function EC_GROUP_get_params is added which creates an OSSL_PARAM array from a given EC_GROUP ptr. It forwards the call to ossl_ec_group_todata which is internal API.

In our project we need to convert an EC_GROUP back to an OSSL_PARAM array to modify it
(in this case shift the generator/base point of the group). 

I was a bit unsure about the naming of the method, should it be EC_GROUP_get_params or EC_GROUP_to_params? There was another function called EC_GROUP_get_ecparameters, so I've chosen _get_.

Also should there be the possibility to pass in an BN_CTX* as an optional parameter? 

Thanks a lot and best regards
Oliver

### CODE REVIEW

**Comment by mattcaswell** [comment]
Also please run "make update" and check in the resulting changes.

File: crypto/ec/ec_lib.c
Comment (Basskrapfen) [reply crypto/ec/ec_lib.c]: Renamed function EC_GROUP_to_params. Thanks!

File: crypto/ec/ec_lib.c
Comment (Basskrapfen) [reply crypto/ec/ec_lib.c]: Added optional BN_CTX parameter.

**Comment by Basskrapfen** [comment]
> Please add some documentation for the new function

@mattcaswell 
Where is the best place to put more documentation? Just as a comment of the function?

**Comment by t8m** [comment]
> EC_GROUP_new.pod

That would be the right place for the documentation.

File: include/openssl/ec.h
Comment (t8m) [inline include/openssl/ec.h:474]: You can drop this comment once you add the proper documentation into the .pod file.

File: test/ectest.c
Comment (t8m) [inline test/ectest.c:2074]: Please use the classic C-style comments only.

File: test/ectest.c
Comment (t8m) [inline test/ectest.c:2095]: Please split the lines over 80 characters if possible.

File: test/ectest.c
Comment (t8m) [inline test/ectest.c:2154]: Drop the extra empty line

**Review by t8m — CHANGES_REQUESTED** [review]

File: include/openssl/ec.h
Comment (Basskrapfen) [reply include/openssl/ec.h:474]: I've kept it for now to be consistent with the other function in the same file. I can also remove it if you think it is redundant information.

File: test/ectest.c
Comment (Basskrapfen) [reply test/ectest.c]: Changed it, thanks for the hint.

File: test/ectest.c
Comment (Basskrapfen) [reply test/ectest.c]: Splitted all lines to keep the 80 characters limit.

File: test/ectest.c
Comment (Basskrapfen) [reply test/ectest.c:2168]: Removed it, thanks for the hint.

File: doc/man3/EC_GROUP_new.pod
Comment (mattcaswell) [inline doc/man3/EC_GROUP_new.pod:120]: Perhaps worth mentioning that the caller is responsible for freeing the returned `OSSL_PARAM` array

File: include/openssl/ec.h
Comment (mattcaswell) [inline include/openssl/ec.h:467]: Nit: One too many spaces before "pointer"

File: include/openssl/ec.h
Comment (mattcaswell) [inline include/openssl/ec.h:472]: Repeated text on this line

File: test/ectest.c
Comment (mattcaswell) [inline test/ectest.c:2058]: OSSL_PARAM not OSSL_PARAMETERS

File: test/ectest.c
Comment (mattcaswell) [inline test/ectest.c:2125]: Probably should be using `TEST_size_t_gt` here

File: test/ectest.c
Comment (mattcaswell) [inline test/ectest.c:2129]: and here

**Review by mattcaswell — CHANGES_REQUESTED** [review]

File: include/openssl/ec.h
Comment (Basskrapfen) [reply include/openssl/ec.h:472]: Thanks, fixed

File: include/openssl/ec.h
Comment (Basskrapfen) [reply include/openssl/ec.h:467]: woah, good find! :)

File: test/ectest.c
Comment (Basskrapfen) [reply test/ectest.c:2058]: fixed

File: test/ectest.c
Comment (Basskrapfen) [reply test/ectest.c:2125]: Good one, replaced with TEST_size_t_gt, thanks.

File: test/ectest.c
Comment (Basskrapfen) [reply test/ectest.c:2129]: also fixed

**Review by mattcaswell — APPROVED** [review]
Approved subject to the CIs passing. Needs a second review

**Comment by Basskrapfen** [comment]
There seems to be an issue with the win-10-x86 build check which most likely isn't caused by this change:
"[WinError 5] Access is denied: 'E:\\workers\\ossl-win10-x86_64-7\\master_windows-win10-x86\\build\\apps\\libcrypto-3.dll'"

**Comment by mattcaswell** [comment]
> There seems to be an issue with the win-10-x86 build check which most likely isn't caused by this change:
"[WinError 5] Access is denied: 'E:\workers\ossl-win10-x86_64-7\master_windows-win10-x86\build\apps\libcrypto-3.dll'"

We can ignore that one. An occasional intermittent failure that we haven't quite got to the bottom of.

**Review by t8m — APPROVED** [review]

**Comment by mattcaswell** [comment]
Setting to "ready to merge" but repo is still frozen for the release.

**Comment by tmshort** [comment]
Merged to the `master` branch. Thank you for your contribution!

**Pull request closed without merging**
